tests/js/client/shell/shell-debug-dump.js

```diff
@@ -326,11 +329,86 @@ function debugDumpGraphTestsuite () {
   };
 }
 
+function debugDumpViews () {
+  return {
+    setUp: function () {
+      db._createDatabase(dbName, { replicationFactor: 3, writeConcern: 2 });
+      db._useDatabase(dbName);
+
+      analyzers.save("my_delimiter", "delimiter", {"delimiter": "@"});
+      analyzers.save("delimiter_hyphen", "delimiter", {"delimiter": "-" });
+      analyzers.save("text_en_nostem", "text", { locale: "en", case: "lower", accent: false, stemming: false, stopwords: []});
+      
+      db._create(cn, {numberOfShards: 3});
+      db._create(cn1, {numberOfShards: 3});
+
+
+      for (let i = 0; i < 50; ++i) {
+        db[cn].insert({ value1: i + "@" + i, value2: i});
+        db[cn1].insert({ value3: i + "@" + i, value4: i});
+      }
+
+      let properties = {"links": {}};
+      properties["links"][cn] = {
+        "analyzers": [
+            "my_delimiter",
+            "identity",
+            "text_en_nostem"
+        ],
+        "fields": {
+            "value1": {},
+            "value2": {}
+        }
+      };
+      properties["links"][cn1] = {
+        "analyzers": [
+          "my_delimiter",
+        ],
+        "fields": {
+            "value3": {},
+            "value4": {}
+        }
+      };
+      db._createView("view", "arangosearch", properties);
+    },
+
+    tearDown: function () {
+      db._useDatabase("_system");
+      db._dropDatabase(dbName);
+      try {
+        fs.unlink(fileName);
+      } catch (err) {
+      }
+      try {
+        fs.unlink(outFileName);
+      } catch (err) {
+      }
+    },
+
+    testDebugDumpWithViews: function () {
+      let query = `for d in view search ANALYZER(d.value1 == "31" or d.value1 == "1", "my_delimiter") return d `;
```

Comment:
Shouldn't this query use waitForSync? Otherwise it is not guaranteed that all values from the inserts in `setUp` have made it into the view.
```suggestion
      let query = `for d in view search ANALYZER(d.value1 == "31" or d.value1 == "1", "my_delimiter") OPTIONS {waitForSync: true } return d `;
```